### ISSUE TITLE
Review lab_2 by Federico Buccellato s309075 

### ISSUE THREAD

**Issue by FedeBucce** [theMiriam/Computational_intelligence#1]
Hi Miriam,
In general, I like the idea of your code. 
The only things I would like to point out are the following:
- The code is very slow in producing results, and in my opinion, the win rate could be increased with some improvements in "fitness" management. 
I'm not entirely clear on why you have the command "strategy = (optimal, optimal)"  in the "fitness" function. 
In this way, you no longer use the various strategies you created for adaptive_strategies.  
Perhaps it was just a moment of distraction.
(I tried to remove it, your code actually achieves a much higher win rate)

In conclusion, I would say that, in my opinion, the foundation of your code is very good, and with some adjustments, it can be significantly improved.

I hope this review was useful to you and good luck on your next labs.
